cmd/daemon/main.go

```diff
@@ -413,20 +392,8 @@ func (d *daemon) make(c *cli.Context) error {
 		return err
 	}
 
-	errCh := s.Start()
-	go func() {
-		select {
-		case <-d.ctx.Done():
-			return
-		case err := <-errCh:
-			log.Errorf("failed to start RPC server: %s", err)
-			d.cancel()
-			os.Exit(1)
-		}
-	}()
```

Comment:
`Start()` now blocks and it internally uses the context that was passed to `rpcCfg`.